File: benchmarks/execution/Rust_embedded/Train_1_beebook_deterministic_exec.rs
#![ allow( dead_code, unused, non_snake_case, non_camel_case_types, unused_assignments ) ]
//#![no_std] //std is enabled for the generated machine by default, since without it there is no println which makes unit-testing very difficult
use btypes::{bset, brel};
use btypes::bset::BSet;
use btypes::bset::SetItem;
use btypes::bset::PowSetItem;
use btypes::bset::PowAble;
use btypes::bset::NestedSet;
use btypes::brelation::BRelation;
use btypes::brelation::RelLeftItem;
use btypes::brelation::RelPowAble;
use btypes::bboolean::BBoolean;
use btypes::bboolean::BBool;
use btypes::bboolean::BOOL;
use btypes::binteger::BInteger;
use btypes::binteger::BInt;
use btypes::binteger::set_BInteger;
use btypes::btuple::BTuple;


mod Train_1_beebook_deterministic;




#[derive(Default, Debug)]pub struct Train_1_beebook_deterministic_exec {
    /*declarations*/counter: BInteger,
    /*includes*/_Train_1_beebook_deterministic: Train_1_beebook_deterministic::Train_1_beebook_deterministic,
}


impl Train_1_beebook_deterministic_exec {

    pub fn new() -> Train_1_beebook_deterministic_exec {
        //values: ''
        let mut m: Train_1_beebook_deterministic_exec = Default::default();
        m.init();
        return m;
    }
    fn init(&mut self) {
        /*set_initializations*/
        /*properties*/
        /*body*/self.counter = 0;
        /*includesInitialization*/self._Train_1_beebook_deterministic = Train_1_beebook_deterministic::Train_1_beebook_deterministic::new();
    }

    pub fn get_counter(&self) -> BInteger {
        return self.counter.clone();
    }

    pub fn simulate(&mut self) -> () {
        while self.counter.less(&10000) {
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R4);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R4);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R4);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R4);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::M);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::M);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R5);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R5);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::K);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R5);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::K);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::F);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::F);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R5);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::M);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::M);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::J);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::G);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R4);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::J);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R9);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::N);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R9);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R9);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R5);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R9);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::G);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::G);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::F);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.route_reservation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.point_positionning(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.route_formation(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::F);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R1);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::K);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.FRONT_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.FRONT_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::K);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::I);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::H);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::C);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::B);
            self._Train_1_beebook_deterministic.BACK_MOVE_2(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::A);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::L);
            self._Train_1_beebook_deterministic.BACK_MOVE_1(/*enum_call*/Train_1_beebook_deterministic::BLOCKS::M);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R6);
            self._Train_1_beebook_deterministic.route_freeing(/*enum_call*/Train_1_beebook_deterministic::ROUTES::R9);
            self.counter = self.counter.plus(&1);
        }
    }
}

fn main() {
    let mut exec = Train_1_beebook_deterministic_exec::new();
    exec.simulate();
}

File: benchmarks/execution/C++_Primitives/Train_1_beebook_deterministic_exec.cpp
#include <iostream>
#include <string>
#include <btypes_primitives/BUtils.hpp>
#include <btypes_primitives/StateNotReachableError.hpp>
#include <btypes_primitives/PreconditionOrAssertionViolation.hpp>
#include <btypes_primitives/BBoolean.hpp>
#include <btypes_primitives/BObject.hpp>
#include <btypes_primitives/BInteger.hpp>
#include "Train_1_beebook_deterministic.hpp"

#ifndef Train_1_beebook_deterministic_exec_H
#define Train_1_beebook_deterministic_exec_H

using namespace std;

class Train_1_beebook_deterministic_exec {

    public:



    private:


        Train_1_beebook_deterministic _Train_1_beebook_deterministic;


        BInteger counter;
    public:
        Train_1_beebook_deterministic_exec() {
            counter = (BInteger(0));
        }

        void simulate() {
            while((counter.less((BInteger(10000)))).booleanValue()) {
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R4)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R4)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R4)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R4)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::M)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::M)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R5)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R5)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::K)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R5)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::K)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::F)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::F)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R5)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::M)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::M)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::J)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::G)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R4)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::J)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R9)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::N)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R9)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R9)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R5)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R9)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::G)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::G)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::F)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.route_reservation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.point_positionning((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.route_formation((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::F)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R1)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::K)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_1((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.FRONT_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::K)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::I)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::H)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::C)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::B)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_2((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::A)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::L)));
                this->_Train_1_beebook_deterministic.BACK_MOVE_1((Train_1_beebook_deterministic::BLOCKS(Train_1_beebook_deterministic::BLOCKS::M)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R6)));
                this->_Train_1_beebook_deterministic.route_freeing((Train_1_beebook_deterministic::ROUTES(Train_1_beebook_deterministic::ROUTES::R9)));
                counter = counter.plus((BInteger(1)));
            }
        }

        BInteger _get_counter() const {
            return counter;
        }

        friend std::ostream& operator<<(std::ostream &strm, const Train_1_beebook_deterministic_exec &machine) {
          strm << "_Train_1_beebook_deterministic: " << machine._Train_1_beebook_deterministic << "\n";
          strm << "_get_counter: " << machine._get_counter() << "\n";
          return strm;
        }

};

int main() {
    Train_1_beebook_deterministic_exec exec;
    exec.simulate();
    cout << exec << "\n";
    return 0;
}

#endif